src/layer/types.js

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import * as React from 'react';
+import {PLACEMENT} from './constants.js';
+
+/** LayersManager */
+export type LayersManagerPropsT = {
+  children: React.Node,
+};
+
+export type LayersContexT = {
+  root: typeof undefined | ?HTMLElement,
+  host: typeof undefined | ?HTMLElement,
+};
+
+/** Layer */
+export type LayerPropsT = {
+  children: React.Node,
+  host?: ?HTMLElement,
+  index?: number,
+  mountNode?: HTMLElement,
+  onMount?: () => mixed,
+  onUnmount?: () => mixed,
+};
+
+export type LayerComponentPropsT = {
+  children: React.Node,
+  host: ?HTMLElement,
+  index?: number,
+  mountNode?: HTMLElement,
+  onMount?: () => mixed,
+  onUnmount?: () => mixed,
+};
+
+export type LayerStateT = {
+  container: ?HTMLElement,
+};
+
+/** TetherBehavior */
+export type PopperPlacementT = $Keys<typeof PLACEMENT>;
```

Comment:
It may be worth re-naming these types to remove references to popover so that it is more generalized.

Reply:
renamed in the Tether PR